src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs

```diff
@@ -318,7 +318,10 @@ public void Dispose()
             }
             if (_intermediateFolder != null)
             {
-                // to-do: add check for build status. if failed, not overwrite buildinfo
+                if (_currentBuildInfo.Versions.Any(v => v.Status == BuildStatus.Failed))
```

Comment:
This shuold work for OP as it can provide the latest succeeded output. Will it break for local if output and intermediate file don't match?